lib/src/license_detection/license_detector.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:pana/src/license_detection/confidence.dart';
+import 'package:pana/src/license_detection/license.dart';
+import 'package:pana/src/license_detection/primary_filter.dart';
+import 'package:pana/src/license_detection/token_matcher.dart';
+
+// Load corpus licenses.
+final _licenses = loadLicensesFromDirectories(_directories);
+
+// WIP: Returns a list of detected licenses whose
+// confidence score is above a certain threshold.
+List<LicenseMatch> detectLicense(String text, double threshold) {
+  final granularity = computeGranularity(threshold);
+  final unknownLicense =
+      LicenseWithNGrams.parse(License.parse('', text), granularity);
+  final possibleLicenses =
+      filter(unknownLicense.occurrences, _licenses, granularity);
+  var result = <LicenseMatch>[];
+
+  for (var license in possibleLicenses) {
+    final matches = findPotentialMatches(
+      unknownLicense,
+      license,
+      threshold,
+      granularity,
+    );
+
+    for (var match in matches) {
+      final licenseMatch =
+          confidenceMatch(unknownLicense, license, match, threshold);
+
+      if (licenseMatch != null) {
+        result.add(licenseMatch);
+      }
+    }
```

Comment:
```suggestion
    for (var match in matches) {
      try {
        final licenseMatch =
            confidenceMatch(unknownLicense, license, match, threshold);
         result.add(licenseMatch);
      } on LicenseMismatchException {
         // Do nothing, because this was a mismatch
      }
     }
```